unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BlockerController/BlockerAnimationHandler.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections;
+using DCL.Helpers;
+using UnityEngine;
+
+namespace DCL.Controllers
+{
+    public interface IBlockerAnimationHandler
+    {
+        void FadeIn(GameObject go);
+        void FadeOut(GameObject go, System.Action OnFinish);
+    }
+
+    /// <summary>
+    /// This class is in charge of handling the FadeIn/FadeOut animation
+    /// of blockers.
+    /// </summary>
+    public class BlockerAnimationHandler : IBlockerAnimationHandler
```

Comment:
I have seen that this class could be used for any game object with a `Renderer`. Maybe we could separate it from the blockers folders, rename it and set it as a general tool for any time where we need to apply fade in/out to any other game object? What do you think?

Reply:
Mmmh the issue I see with this approach is that this class is expecting the Blocker prefab arrangement/shader, so we can end up re-using it and then when we need to modify the code to change blocker-specific stuff, the re-used instances will break. For instance, we may add a new specific property to the blocker shader and we might want to animate it here.

I could've used animations for this, but decided for this lightweight approach because the blockers are widely used.

I propose to keep this as it is and if in the future we need to apply FadeIn/FadeOut to another non-ui object we can make this more generic, or maybe reuse the interface and change its name.

Reply:
Ok! I agree with that!